js/PixelPainter.js

```diff
@@ -0,0 +1,387 @@
+      paletteRow.appendChild(swatch);
+
+      swatch.addEventListener('click', setPaintbrush);
+    }
+  }
+}
+
+//Default palette onload
+createRandomPalette(4, 15);
+
+paletteGeneratorButton.addEventListener('click', function () {
+  randomPixelPalette.remove();
+  createRandomPalette(pixelx, pixely);
+})
+
+//Create canvas
+const createCanvas = function (depth, width = depth) {
+  const canvas = document.createElement('div');
+  canvas.id = 'pixelCanvas';
+  canvas.style.width = (15 * depth) + 'px';
+  createContainer1.appendChild(canvas);
+
+  for (let y = 0; y < width; y++) {
+    let pixelRow = document.createElement('div');
+    pixelRow.className = 'pixelRow'
+    canvas.appendChild(pixelRow);
+
+    for (let x = 0; x < depth; x++) {
+      let pixel = document.createElement('div');
+      pixel.className = 'pixel';
+      pixel.dataset.column = x;
+      pixel.dataset.row = y;
+      pixel.dataset.fillValue = false;
+      pixelRow.appendChild(pixel);
+
+      pixel.addEventListener('mousedown', setPixelColor);
+      pixel.addEventListener('mouseup', releaseColor);
+      pixel.addEventListener('mouseover', paintColor);
+    }
+  }
+}
+
+//Default canvas onload
+createCanvas(50, 25);
+
+let canvas = document.querySelector('#pixelCanvas');
+
+//Create input img url
+const createImgDiv = document.createElement('div');
+createImgDiv.id = 'customImgDiv';
+document.body.appendChild(createImgDiv);
+
+const imgInputDiv = document.createElement('div');
+imgInputDiv.id = 'imgInputDiv';
+createImgDiv.appendChild(imgInputDiv);
+
+const createImgInput = document.createElement('input');
+createImgInput.id = 'customImgInput';
+createImgInput.type = 'text';
+imgInputDiv.appendChild(createImgInput);
+
+const imgInputDirections = document.createElement('div');
+imgInputDirections.id = 'imgInputWords';
+imgInputDirections.innerHTML = 'Enter URL and click here for custom image';
+createImgDiv.appendChild(imgInputDirections);
+
+imgInputWords.addEventListener('click', changeCanvasImg);
+
+function changeCanvasImg() {
+  pixelCanvas.style.backgroundImage = "url('" + customImgInput.value + "')";
+}
+
+//Save to local storage
+function savePixelCanvas() {
+  canvas = document.querySelector('#pixelCanvas');
+  localStorage.clear();
+  let key = 'yourPixelPainterKey';
+  let value = canvas.innerHTML;
+  localStorage.setItem(key, value);
+  localStorage.setItem('canvasx', canvasx);
+  localStorage.setItem('canvasy', canvasy);
+  localStorage.setItem('pixelx', pixelx);
+  localStorage.setItem('pixely', pixely);
+}
+
+//Load from local storage
+function loadLocalStorage() {
+  let reload = localStorage.getItem('yourPixelPainterKey');
+  let reloadx = localStorage.getItem('canvasx');
+  let reloady = localStorage.getItem('canvasy');
+  let reloadpx = localStorage.getItem('pixelx');
+  let reloadpy = localStorage.getItem('pixely');
+
+  //recreate pixel canvas
+  createRandomPalette(reloadpx, reloadpy);
+  createCanvas(reloadx, reloady);
```

Comment:
This seem unnecessary since it's creating a grid, then you're blowing it away with the line `canvas.innerHTML = reload`. You could just create the canvas div element here and append it to the right location and achieve the same effect.